js_modules/dagster-ui/packages/ui-core/src/runs/RunsRoot.tsx

```diff
@@ -209,50 +211,59 @@ export const RunsRoot = () => {
           }}
         >
           {({pipelineRunsOrError}) => {
-            if (pipelineRunsOrError.__typename !== 'Runs') {
+            function Wrapper() {
```

Comment:
Created this wrapper component so that I can use `useLayoutEffect`

Reply:
It's scary to have a Wrapper component that is defined inside this component because it'll tear down and re-mount on every render rather than just updating (Since no two "<Wrapper />" will ever be the same across renders)

Can you make this a normal component?

Also as a side note, this code seems to be measuring the combined time of loading + initial render? We already have the API timings in datadog, would it be more interesting to start the trace in a useMemo and end it in a useLayoutEffect to capture the render time only? 

If we just wanted to measure render perf, we could put the timings inside the RunsTable and not in this parent component.

Reply:
We want to capture E2E perf, so I'm actually going to use the endTimestamp only. Down the line we could use the start timestamp for constructing exact spans